File: AdminArea/AdminArea.Web/Modules/ServerTypes/Reviewcom/ReviewcomColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { ReviewcomRow } from "./ReviewcomRow";

export interface ReviewcomColumns {
    Id: Column<ReviewcomRow>;
    Name: Column<ReviewcomRow>;
    Email: Column<ReviewcomRow>;
    Subject: Column<ReviewcomRow>;
    Message: Column<ReviewcomRow>;
}

export class ReviewcomColumns extends ColumnsBase<ReviewcomRow> {
    static readonly columnsKey = 'Reviewcom.Reviewcom';
    static readonly Fields = fieldsProxy<ReviewcomColumns>();
}

File: AdminArea/AdminArea.Web/Modules/Reviewcom/Reviewcom/ReviewcomColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace AdminArea.Reviewcom.Columns;

[ColumnsScript("Reviewcom.Reviewcom")]
[BasedOnRow(typeof(ReviewcomRow), CheckNames = true)]
public class ReviewcomColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string Name { get; set; }
    public string Email { get; set; }
    public string Subject { get; set; }
    public string Message { get; set; }
}